### PR TITLE
Support new bloganuary card. Enable bloganauary automatically in January

### PR DESCRIPTION
**Pull request by roo2**
Supports showing the bloganuary my home card to more users and having it dismissed separately from the blogging prompts card. 

Discussed here pe7F0s-1iI-p2#comment-1840.

Screen Shot 2023-12-14 at 12 46 07 pm

## Testing Instructions

Apply diff D131976-code and you'll have to modify that diff to simulate it being January on the backend.

On the front end, apply `?flags=+bloganuary`.

You should see the blogging prompts card skinned for bloganuary.

You should be able to dismiss the card.

If "Bloganuary" is turned off, it should revert to showing the blogging prompts card if the site_intent===write. If the bloganuary card was dismissed, the blogging-prompts card will not be dismissed.

The card can be un-dismissed like so: `wp user-attributes undismiss_a_card --user_id=182004064 --blog_id=211923291 --card=home-section-bloganuary-blogging-prompt`

### CODE REVIEW

File: client/data/blogging-prompt/is-bloganuary.ts [hunk 1]
```diff
@@ -1,9 +1,8 @@
 import { isEnabled } from '@automattic/calypso-config';
-
+import moment from 'moment';
 /**
- * In future this will be automatically enabled in January. For now it just checks a feature flag.
  * @returns true if bloganuary mode is active
  */
 export default function isBloganuary() {
-	return isEnabled( 'bloganuary' );
+	return moment( '2020-01-12' ).format( 'MM' ) === '01' || isEnabled( 'bloganuary' );
```
Comment (eoigal) [inline client/data/blogging-prompt/is-bloganuary.ts]: I think this will always return true. Is this in for testing purposes?

**Review by Addison-Stavlo — COMMENTED** [review]
I am having issues testing this.

Im am applying the diff (with the is_bloganuary function forced to return true), sandboxing public-api and the test sites, and testing the front-end with the bloganuary flag.

Im not seeing any card on old test sites that are not `write` intent. And my site that would normally show a daily prompt card is showing none.

File: client/data/blogging-prompt/is-bloganuary.ts (hunk 1 above)
Comment (roo2) [reply client/data/blogging-prompt/is-bloganuary.ts]: 🤦 oops I didn't mean to push that! 😂

**Comment by roo2** [comment]
> Im not seeing any card on old test sites that are not write intent. And my site that would normally show a daily prompt card is showing none.

hmmm I'll retest again this morning with a clean sandbox/calypso and double check.

**Comment by roo2** [comment]
> Im not seeing any card on old test sites that are not write intent. And my site that would normally show a daily prompt card is showing none.

Ahh yes, it was an error in D131976-code. I had missed the `VIEW_SITE_SETUP` view, so any sites unlaunched would not show the card (or the original blogging prompt card). I've added SECTION_BLOGANUARY_BLOGGING_PROMPT to that view now in  D131976-code.

**Comment by roo2** [comment]
I've updated this PR and the backeend diff and retested, the bloganuary card should now be shown to pre and post launch sites, there are other card layouts where I didn't add it to, such as the expiring domain notice, but from what I've seen those other views are mostly used when there is an urgent event (such as expiring domain) that would take precendence. The vast majority of sites should be using either `VIEW_POST_LAUNCH` or `VIEW_SITE_SETUP` which I've added the bloganuary card to

**Review by Addison-Stavlo — COMMENTED** [review]
> I had missed the VIEW_SITE_SETUP view,

Ah this must have been it for me.

A distinction, these sites for me **were launched**. But the site setup checklist in my home is still present until all tasks are completed or skipped. So my launched sites were still falling under this view.

Testing now I see the card on the site setup view, but after dismissing/completing the last remaining items in site setup checklist and that view ending I no longer see the bloganuary card.

Aside from that my launched write site that normally shows the prompt/bloganuary card seems to show no card at all when running this diff with is_bloganuary returning `true` and the flag.

**Comment by Addison-Stavlo** [comment]
> Testing now I see the card on the site setup view, but after dismissing/completing the last remaining items in site setup checklist and that view ending I no longer see the bloganuary card.

> Aside from that my launched write site that normally shows the prompt/bloganuary card seems to show no card at all when running this diff with is_bloganuary returning true and the flag.

Huh. After re-building calypso locally, re-patching the diff. etc. (all the same things, just an hour or so later).... On the first site the card is back where I would expect it to be post site setup. The second site shows the card as expected. So now all seems to be where it should...

@eoigal - can you try giving this a test to see if it works as expected on your end as well? It has been flaky for me for some reason and another test would be helpful.

**Comment by Addison-Stavlo** [comment]
> but after dismissing/completing the last remaining items in site setup checklist and that view ending I no longer see the bloganuary card.

Speaking with @roo2, he mentioned there was a celebration view that lasts for 30 minutes. That would explain why it disappeared for me and re-appeared when I circled back a bit later. This seems fine.

Im unsure about my other site being inconsistent with showing the Bloganuary card from this. It ended up being there in the end, I say if it tests well to someone else then ship it. 🚢

**Review by eoigal — APPROVED** [review]
Wfm

**Pull request merged**